Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:9.11.1
+
+RUN apt-get update && apt-get install -y rsync
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+RUN npm install
+
+COPY . /usr/src/app/
+
+RUN npm run build
+
+CMD [ "npm", "start" ]
```

Comment:
IMHO we should separate the npm build and install processes into a multi-stage Dockerfile and have a lean runtime docker image with only the the bare essentials that our service needs at runtime.
We could even reconsider going with having `--only=prod` and `--only=dev` deps installed in different images so that the runtime only has prod ones in the end.

Reply:
In the future, we can make some small modifications to the node so you can `npm i -g poet-node`. Packages published to NPM are already "built" so you don't need to `npm run build`, and when installing globally (`-g`), it's made available as a script (you can then run `poet-node` from any location) and has all its dependencies. 
That way we could have the release docker image just do `npm i -g poet-node` and skip all building etc.

Reply:
Actually, it doesn't have to be "in the future", it's an small change which we could address right now if you think it makes sense.

Reply:
If we could have `poet-node` as a npm package that one would just install and run the service, that would be awesome and it would help us do the `frost-api` docker image in a more cleaner and sane way.
So if it requires a small change and not much effort to make it happen, I'd say let's do it.

Reply:
How does installing a package that's published on npm help with testing code that has not been published yet? My assumption is that this docker container is being built to run tests and deploy internally. All of that needs to happen before publishing to npm. Is that correct?

Reply:
It would definitely help with building pipelines for testing some of our code that depends and relies upon `poet-node` (eg `frost-api`) since we're running the same set of commands, installing and building `node`.
I do not mind building either `poet-node` docker image or a simple package as an artifact that I could use for other purposes as well. So here we should focus ourselves on setting up the pipelines for testing and building `poet-node` artifact, whatever that might be.

Reply:
See https://github.com/poetapp/node/pull/103.

Reply:
agreed. It feels like we are tackling too many steps at once. Let's get the pipeline for testing and building the package in place first. That way we can all play around with it and make sure we've built it correctly before building on top of it.

Reply:
Agreed. I think this should be a separate discussion — this PR is dockerizing Node. In this scope and contexts, it makes more sense to have the container grab the local sources and build them.

Other projects that consume/interact with the node can use the published package.